### PR TITLE
Adding GeminiToolThread and same way as c-s stress is running

### PR DESCRIPTION
**Pull request by aleksbykov**
Refactor gemini thread code to use
 - New class GeminiToolThread for run, verify gemini tool
 - Use ThreadPoolExecutor to run gemini command
 - remove some methods from BaseLoaderSet for gemini
 - Adding new GeminiToolEvent to SctEvents

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (`hydra unit-tests`)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
@fruch hope that understood  you correctly.
1. if gemini command terminated with error (exception) the type of event automatically changed to error
2. Adding all events which gemini reports as events

gemini command terminated by on first error with exit code > 0.

**Comment by bentsi** [comment]
@fruch please review

File: sdcm/sct_events.py [hunk 1]
```diff
@@ -253,6 +253,31 @@ def __str__(self):
         return self.msg.format(super(FullScanEvent, self).__str__(), self)
 
 
+class GeminiToolEvent(SctEvent):
```
Comment (bentsi) [inline sdcm/sct_events.py]: `GeminiToolEvent` -> `GeminiEvent`

**Comment by bentsi** [comment]
please resolve conflicts

File: sdcm/gemini_thread.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+                result = node.remoter.run(cmd=gemini_cmd,
+                                          timeout=self.timeout,
+                                          ignore_status=False,
+                                          log_file=log_file_name)
+                # sleep to gather all latest log messages
+                time.sleep(5)
+        except Exception as details:
+            LOGGER.error(details)
+            result = details.result
```
Comment (fruch) [inline sdcm/gemini_thread.py]: can be more Exception here would rather pick the result with `result = getattr(details, 'resault', 'n/a')` or something like that

File: sdcm/gemini_thread.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        "WARN": "WARNING",
+        "ERROR": "ERROR",
+        "FATAL": "CRITICAL"
+    }
+
+    def __init__(self, node, gemini_log_filename):
+        super(GeminiToolEventsPublisher, self).__init__()
+        self.gemini_log_filename = gemini_log_filename
+        self.node = str(node)
+        self.gemini_events = [GeminiToolLogEvent(type='geminievent', regex=r'{"L":"(?P<type>[A-Z]+?)".+"M":"(?P<message>[\w\s.,]+?)"}', severity=Severity.CRITICAL)]
```
Comment (fruch) [inline sdcm/gemini_thread.py]: if `GeminiToolEvent` changes to `GeminiToolEvent`
so `GeminiToolLogEvent` should also change to `GeminiLogEvent`

**Comment by bentsi** [comment]
@aleksbykov please resolve conflicts and address comments

File: sdcm/sct_events.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/sct_events.py]: done

File: sdcm/gemini_thread.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/gemini_thread.py]: renamed

File: sdcm/gemini_thread.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/gemini_thread.py]: fixed

**Comment by aleksbykov** [comment]
@fruch @bentsi Resolved

**Comment by bentsi** [comment]
Pelase put "Ready for review" label when ready :)
image 
when

**Pull request merged**